public/app/features/dashboard-scene/settings/GeneralSettings.tsx

```diff
@@ -22,11 +69,190 @@ export class GeneralSettingsEditView
   static Component = ({ model }: SceneComponentProps<GeneralSettingsEditView>) => {
     const dashboard = getDashboardSceneFor(model);
     const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+    const { title, description, tags, meta, editable, graphTooltip, overlay } = dashboard.useState();
+    const timeRange = sceneGraph.getTimeRange(dashboard);
+    const refreshPicker = getRefreshPickerControl(dashboard);
+
+    const onTitleChange = (value: string) => {
+      dashboard.setState({ title: value });
+    };
+
+    const onDescriptionChange = (value: string) => {
+      dashboard.setState({ description: value });
+    };
+
+    const onTagsChange = (value: string[]) => {
+      dashboard.setState({ tags: value });
+    };
+
+    const onFolderChange = (newUID: string, newTitle: string) => {
+      const newMeta = {
+        ...meta,
+        folderUid: newUID || meta.folderUid,
+        folderTitle: newTitle || meta.folderTitle,
+        hasUnsavedFolderChange: true,
+      };
+
+      dashboard.setState({ meta: newMeta });
+    };
+
+    const onEditableChange = (value: boolean) => {
+      dashboard.setState({ editable: value });
+    };
+
+    const onTimeZoneChange = (value: TimeZone) => {
+      dashboard.state.$timeRange?.setState({
```

Comment:
I suggest adding a new function to retrieve time range from dashboard scene to https://github.com/grafana/grafana/blob/main/public/app/features/dashboard-scene/utils/dashboardSceneGraph.ts, this way - if we ever change where the time range is placed in the hierarchy - we will be able to spot that quickly via that graph function test.